Azure/sap-hana
# Backlog: Copy XSA Key Files

## Problem

Closes #426

Closes #385

## Description

This PR:

- Refactors the `copy_ssfs_keys.yml` into two tasks to simplify and extend usability.
- Adds a new task, `set_runtime_path_facts.yml`, to determine the SAP HANA global path at runtime.
- Removes the role `vars` which are now computed at runtime.

## Pre-Requisites

- A HANA cluster with HSR and XSA.
- Log in to the RTI.

## Commitment to Test

- Testing takes 5 minutes.
- Reviewing changed files.

## Test Instructions / Acceptance Criteria

- [ ] Code changes are acceptable.

Using the RTI, confirm the following:

- [ ] Check HSR: Running `ansible -i ~/hosts 10.1.1.4 -m shell -a "sudo su - hn1adm -c 'hdbnsutil -sr_state'  2>/dev/null | grep '^Operation'" --become 2>/dev/null` reports:

  ```text
  10.1.1.4 | CHANGED | rc=0 >>
  Operation mode of SITEA: primary
  Operation mode of SITEB: logreplay
  ```

For the following checks, ensure the SHA1 keys are the same for each host. Examples are shown, below.

- [ ] Check main SSFS.DAT files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/security/rsecssfs/data/SSFS_HN1.DAT" --become 2>/dev/null | grep SSFS` reports:

  ```text
  e3f42924a072e59d0688b89d437443f0a7fe56c7  /usr/sap/HN1/SYS/global/security/rsecssfs/data/SSFS_HN1.DAT
  e3f42924a072e59d0688b89d437443f0a7fe56c7  /usr/sap/HN1/SYS/global/security/rsecssfs/data/SSFS_HN1.DAT
  ```

- [ ] Check main SSFS.KEY files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY" --become 2>/dev/null | grep SSFS` reports:

  ```text
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  ```
- [ ] Check XSA SSFS.DAT files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/xsa/security/ssfs/data/SSFS_HN1.DAT" --become 2>/dev/null | grep SSFS` reports:

  ```text
  bc3f37f75165e98697ae7e558e8136b17d57ba96  /usr/sap/HN1/SYS/global/xsa/security/ssfs/data/SSFS_HN1.DAT
  bc3f37f75165e98697ae7e558e8136b17d57ba96  /usr/sap/HN1/SYS/global/xsa/security/ssfs/data/SSFS_HN1.DAT
  ```

- [ ] Check XSA SSFS.KEY files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/xsa/security/ssfs/key/SSFS_HN1.KEY" --become 2>/dev/null | grep SSFS` reports:

  ```text
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  ```
